src/lib.rs

```diff
@@ -210,3 +223,191 @@ impl fmt::Display for Error {
 }
 
 impl std::error::Error for Error {}
+
+/// Run the provided actor.
+///
+/// This is the primary event loop of an actor which takes messages out of the mailbox and hands
+/// them to the actor.
+pub async fn run<A>(mut mailbox: Mailbox<A>, mut actor: A) -> A::Stop
+where
+    A: Actor,
+{
+    if let Err(stop) = actor.started(&mut mailbox).await {
+        return stop;
+    }
+
+    while let ControlFlow::Continue(()) = yield_once(&mut mailbox, &mut actor).await {}
+
+    actor.stopped().await
+}
+
+/// Yields to the manager to handle one message, returning the actor should be shut down or not.
+pub async fn yield_once<A>(mailbox: &mut Mailbox<A>, actor: &mut A) -> ControlFlow<(), ()>
+where
+    A: Actor,
+{
+    let message = mailbox.next().await;
+
+    tick(message, actor, mailbox).await
+}
+
+/// Handle one message and return whether to exit from the manage loop or not.
+///
+/// Note that this will immediately create the message handler span if the `instrumentation`
+/// feature is enabled.
+pub fn tick<'a, A>(
+    message: Message<A>,
+    actor: &'a mut A,
+    mailbox: &'a mut Mailbox<A>,
+) -> TickFuture<'a, A>
+where
+    A: Actor,
+{
+    TickFuture::new(message.0, actor, mailbox)
+}
```

Comment:
I am not yet super happy about this. Do you think we can merge this into `ReceiveFuture` somehow?

Reply:
Perhaps not into `ReceiveFuture` itself but the indirection here and also the `Span` handling is rather confusing IMO. It would be nice if users would just need to call `Mailbox::next`, get a message and pass it to the actor.

Reply:
> and pass it to the actor

Isn't this essentially what `tick` is doing?

Reply:
Yeah, what I find unnecessary is that we need to provide a function for that. It seems like such a simple operation that the user should be able to perform (in a custom event loop f.e.) without calling into a free function of xtra.

I believe the reason for this is that `TickFuture` currently encapsulates the matching on the message variants etc. Perhaps we can move this somewhere else.

Reply:
Does tick not also encapsulate adding the actor to the mix? The next message future just takes the mailbox, whereas tick also borrows the actor.

Reply:
Yep. But really the actor is only needed to then dispatch the message received from the mailbox. it is two operations in fact. I have something locally that I can hopefully put up soon :)

Reply:
When you say "it", do you mean receive + tick?

Reply:
Yeah!

To get the message you only need the `Mailbox` and to dispatch it you only need the actor.

Reply:
In any case, I don't think it needs to block this PR!